tools/analyzer_plugin/test/integration/test_bases/server_plugin_contributor_test_base.dart

```diff
@@ -28,6 +29,15 @@ class SourceSelection {
   final String? target;
 
   SourceSelection(this.source, this.offset, this.length, {this.target});
+
+  @override
+  String toString() {
+    return 'SourceSelection ${{
+      'target': target,
+      'offset': offset,
+      'length': length,
+    }}. Preview:\n' + SourceFile.fromString(source.contents.data, url: source.uri).span(offset, offset + length).highlight();
+  }
```

Comment:
Added this during debugging. 

Before: `Instance of 'SourceSelection'`

After:
```
SourceSelection {target: class #FooComponent# extends, offset: 181, length: 12}. Preview:
  ╷
8 │ class FooComponent extends UiComponent2<FooProps> {
  │       ^^^^^^^^^^^^
  ╵
```